go.mod

```diff
@@ -16,6 +16,7 @@ require (
 	github.com/blang/semver v3.5.1+incompatible
 	github.com/containers/image/v5 v5.30.1
 	github.com/coreos/go-oidc/v3 v3.10.0
+	github.com/coreos/go-semver v0.3.1
```

Comment:
We prefer not to add any new dependencies, unless there is strong justification as it brings about it's own maintenance burden. So, could you please remove this dependency and rely only simple regexes, e.g. https://github.com/openshift/oc/blob/e77658e7eaf7dfe310e779ad9b6a74e581a3715c/pkg/cli/admin/release/extract_tools.go#L1246